airbyte-webapp/src/components/StatusIcon/StatusIcon.tsx

```diff
@@ -1,42 +1,40 @@
 import React from "react";
 import styled from "styled-components";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { faCheck, faTimes, faBan, faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
+import { faCheck, faTimes, faBan, faExclamationTriangle, IconDefinition } from "@fortawesome/free-solid-svg-icons";
 
-import PauseIcon from "./components/Pause";
+import PauseIcon from "./PauseIcon";
 
-type IProps = {
-  success?: boolean;
-  warning?: boolean;
-  title?: string;
-  inactive?: boolean;
-  empty?: boolean;
+export type StatusIconStatus = "empty" | "inactive" | "success" | "warning";
+
+interface Props {
   className?: string;
+  status?: StatusIconStatus;
+  title?: string;
   big?: boolean;
   value?: string | number;
-};
+}
 
-const getWidth = (props: IProps) => {
-  if (props.big) {
-    return props.value ? 57 : 40;
-  }
+const getBadgeWidth = (props: Props) => (props.big ? (props.value ? 57 : 40) : props.value ? 37 : 20);
 
-  return props.value ? 37 : 20;
+const _iconByStatus: Record<StatusIconStatus, IconDefinition | undefined> = {
+  empty: faBan,
+  inactive: undefined,
```

Comment:
This was done to please TypeScript. Is there a way I can avoid it?

Reply:
I kept trying to find some sort of PartialKey thing, but:
`const _iconByStatus: Partial<Record<StatusIconStatus, IconDefinition | undefined>> = {`
works just fine